internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlyStoreConfigRepository.java

```diff
@@ -115,6 +121,15 @@ public Optional<StoreConfig> getStoreConfig(String storeName) {
       veniceStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.extractRegularStoreName(storeName);
     }
     StoreConfig config = storeConfigMap.get().get(veniceStoreName);
+    // Lazy fetch. It should happen when the entire store config map is not loaded, and only do ad-hoc fetch
+    if (config == null) {
+      config = accessor.getStoreConfig(veniceStoreName);
+      if (config != null) {
+        storeConfigMap.get().put(config.getStoreName(), config);
+        accessor.subscribeStoreConfigDataChangedListener(veniceStoreName, storeConfigChangedListener);
```

Comment:
I wonder if we can unsubscribe when the store is deleted from this cluster; is it feasible with the current implementation? Does it know if the store is deleted?

Reply:
IIUC, we will need the `StoreConfigAddedOrDeletedChangedListener` for this and that's only used in `refresh`.. Let me see if I can incorporate Gaojie's suggestion to refactor this into next revision